python/whylogs/api/logger/logger.py

```diff
@@ -98,13 +102,22 @@ def log(
 
         # If segments are defined use segment_processing to return a SegmentedResultSet
         if active_schema and active_schema.segments:
-            return segment_processing(active_schema, obj, pandas, row, self._segment_cache)
+            segmented_results = segment_processing(active_schema, obj, pandas, row, self._segment_cache)
+            _extract_common_profile_metadata(
```

Comment:
I think I'd be happier if the return value was assigned to something.

Reply:
This updates the dictionary. I added a comment in the code.

result_set has a metadata property but not a metadata setter. The idea is that we are aggregating information but not creating new instances of the collection on every modification.

Reply:
I might still do `_ = _populate_common_profile_metadate(...)  # I like side-effects :)`